src/compiler/checker.ts

```diff
@@ -19624,7 +19644,7 @@ namespace ts {
             }
 
             // Functions with with an explicitly specified 'void' or 'any' return type don't need any return expressions.
-            if (returnType && maybeTypeOfKind(returnType, TypeFlags.Any | TypeFlags.Void)) {
+            if (returnType && maybeTypeOfKind(returnType, TypeFlags.AnyOrUnknown | TypeFlags.Void)) {

```

Comment:
I think a function returning an explicit `unknown` should probably have return values? Otherwise you should've written `void` or `undefined`, right?

Reply:
Not sure about that. `undefined` is in the domain of `unknown` (just like it is in the domain of `any`) and that's really what should guide us here.

Reply:
Hm. Fair. Do we handle unions with `undefined` in accordance with that, then?

Reply:
We do, but with a few wrinkles. If the return type annotation includes `void`, `any`, or `unknown` we don't require *any* `return` statements. Otherwise, if the return type includes `undefined` we require at least one `return` statement somewhere, but don't require `return` statements to have expressions and allow the end point of the function to be reachable. Otherwise, we require a `return` statement with an expression at every exit point.